spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -49,70 +50,132 @@
-    expect(function.class.name).to eq('moduleb::rb_func_b')
-  end
+      expect(qualified_function).to be_a(Puppet::Functions::Function)
+      expect(qualified_function.class.name).to eq('modulea::hello')
+    end
 
-  it 'cannot load an unqualified function from a module without metadata' do
-    loaders = Puppet::Pops::Loaders.new(environment_for(module_without_metadata))
+    it 'loads a puppet function from a module without a lib directory' do
+      loaders = Puppet::Pops::Loaders.new(environment_for(module_without_lib))
+      modulea_loader = loaders.public_loader_for_module('modulea')
 
-    moduleb_loader = loaders.public_loader_for_module('moduleb')
+      qualified_function = modulea_loader.load_typed(typed_name(:function, 'modulea::hello')).value
 
-    expect(moduleb_loader.load_typed(typed_name(:function, 'rb_func_b'))).to be_nil
-  end
+      expect(qualified_function).to be_a(Puppet::Functions::Function)
+      expect(qualified_function.class.name).to eq('modulea::hello')
+    end
+
+    it 'loads a puppet function in a sub namespace of module' do
+      loaders = Puppet::Pops::Loaders.new(environment_for(module_with_metadata))
+      modulea_loader = loaders.public_loader_for_module('modulea')
+
+      qualified_function = modulea_loader.load_typed(typed_name(:function, 'modulea::subspace::hello')).value
+
+      expect(qualified_function).to be_a(Puppet::Functions::Function)
+      expect(qualified_function.class.name).to eq('modulea::subspace::hello')
+    end
 
-  it 'makes all other modules visible to a module without metadata' do
-    env = environment_for(module_with_metadata, module_without_metadata)
-    loaders = Puppet::Pops::Loaders.new(env)
+    it 'loader does not add namespace if not given' do
+      loaders = Puppet::Pops::Loaders.new(environment_for(module_without_metadata))
 
-    moduleb_loader = loaders.private_loader_for_module('moduleb')
-    function = moduleb_loader.load_typed(typed_name(:function, 'moduleb::rb_func_b')).value
+      moduleb_loader = loaders.public_loader_for_module('moduleb')
 
-    expect(function.call({})).to eql("I am modulea::rb_func_a() + I am moduleb::rb_func_b()")
+      expect(moduleb_loader.load_typed(typed_name(:function, 'rb_func_b'))).to be_nil
+    end
+
+    it 'loader allows loading a function more than once' do
+      env = environment_for(dependent_modules_with_metadata)
+      loaders = Puppet::Pops::Loaders.new(env)
+
+      moduleb_loader = loaders.private_loader_for_module('user')
+      function = moduleb_loader.load_typed(typed_name(:function, 'user::caller')).value
+      expect(function.call({})).to eql("usee::callee() was told 'passed value' + I am user::caller()")
+
+      function = moduleb_loader.load_typed(typed_name(:function, 'user::caller')).value
+      expect(function.call({})).to eql("usee::callee() was told 'passed value' + I am user::caller()")
+    end
+  end
+
+  context "when loading from a module with metadata" do
```

Comment:
I am changing these, but I don't think we should care for things like example titles. It only really matters in performance sensitive situations.